File: app/src/main/java/com/apy/anlibrary/MainActivity.java
package com.apy.anlibrary;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.lifecycle.Observer;

import com.apy.anlibrary.livedata.LiveDataManager;
import com.apy.anlibrary.livedata.LiveDataSampleActivity;
import com.apy.library.net.base.BaseActivity;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        LiveDataManager.mutableLiveData.observe(this, new Observer<String>() {
            @Override
            public void onChanged(String s) {
                Log.e("LiveDataSampleActivity4",s);
            }
        });
    }

    public void click(View view){
        LiveDataManager.mutableLiveData.postValue("123456");
        startActivity(new Intent(this, LiveDataSampleActivity.class));
    }

    @Override
    public void initData() {

    }
}

File: app/src/main/java/com/apy/anlibrary/MainActivity.kt
package com.apy.anlibrary

import android.content.Context
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() ,SharedPreferences.OnSharedPreferenceChangeListener{
    private val sharedPreferences1: SharedPreferences?
        get() {
            var sharedPreferences = getSharedPreferences("test", MODE_PRIVATE)
            return sharedPreferences
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        SharedPreferences
        var sharedPreferences = sharedPreferences1
        sharedPreferences?.edit()?.putString("haha","haha")

    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) {
        if (key != null) {
            Log.e("MainActivity",key+"   "+sharedPreferences?.getString(key,"default"))
        }
    }

    fun click(view: android.view.View) {
        sharedPreferences1?.edit()?.putString("haha","heihei")
    }
}